pkg/detector/detector.go

```diff
@@ -955,7 +958,6 @@ func (d *ResourceDetector) HandlePropagationPolicyCreation(policy *policyv1alpha
 	}
 
 	for _, key := range matchedKeys {
-		d.RemoveWaiting(key)
```

Comment:
I didn't understand why this line was deleted.

Reply:
Because the waiting obj would be removed when reconciling, as above shows

Reply:
Reconciling will add the key again, keeping it here feels fine and easier to understand.

Reply:
> Reconciling will add the key again, keeping it here feels fine and easier to understand.

https://github.com/karmada-io/karmada/blob/7224234a742924d78b25cbcc46eaeaccd442fbb8/pkg/detector/detector.go#L74

The `waitingObject` is a map, so it won't matter if an obj is added multiple times

If you mean it's for better understanding, I will revert it though